### PR TITLE
[flink] Add Shuffle by partition Option to ContinuousFileSplitEnumerator 

### PR DESCRIPTION
**Pull request by wxplovecc**
### Purpose
Add Shuffle by partition Option to ContinuousFileSplitEnumerator  to balance the low bucket num with partition table

Linked issue: close #xxx

### CODE REVIEW

File: paimon-common/src/main/java/org/apache/paimon/CoreOptions.java [hunk 1]
```diff
@@ -310,6 +310,12 @@ public class CoreOptions implements Serializable {
                             "Max split size should be cached for one task while scanning. "
                                     + "If splits size cached in enumerator are greater than tasks size multiply by this value, scanner will pause scanning.");
 
+    public static final ConfigOption<Boolean> SCAN_SHUFFLE_BY_PARTITION =
+            key("scan.shuffle-by-partition")
```
Comment (JingsongLi) [inline paimon-common/src/main/java/org/apache/paimon/CoreOptions.java:314]: `streaming-read.shuffle-by-partition`?
Put this to `FlinkConnectorOptions`?

File: paimon-common/src/main/java/org/apache/paimon/CoreOptions.java [hunk 2]
```diff
@@ -310,6 +310,12 @@ public class CoreOptions implements Serializable {
...
+                    .booleanType()
+                    .defaultValue(false)
```
Comment (JingsongLi) [inline paimon-common/src/main/java/org/apache/paimon/CoreOptions.java:316]: default value can be true.

File: paimon-common/src/main/java/org/apache/paimon/CoreOptions.java (hunk 1 above)
Comment (JingsongLi) [inline paimon-common/src/main/java/org/apache/paimon/CoreOptions.java:314]: Also modify `MonitorFunction`?

File: paimon-common/src/main/java/org/apache/paimon/CoreOptions.java (hunk 2 above)
Comment (wxplovecc) [reply paimon-common/src/main/java/org/apache/paimon/CoreOptions.java]: 1

File: paimon-common/src/main/java/org/apache/paimon/CoreOptions.java (hunk 1 above)
Comment (wxplovecc) [reply paimon-common/src/main/java/org/apache/paimon/CoreOptions.java]: 1

Comment (wxplovecc) [reply paimon-common/src/main/java/org/apache/paimon/CoreOptions.java]: done

**Comment by JingsongLi** [comment]
Seems tests failed.

**Comment by wxplovecc** [comment]
> Seems tests failed.

My mistake, the document wasn't generated.

**Pull request merged**